packages/textInput/components/TextInputWithBadges.tsx

```diff
@@ -119,33 +121,37 @@ export class TextInputWithBadges extends TextInputWithIcon<
             >
               {this.getIconStartContent()}
               {this.props.badges &&
-                this.props.badges.map(badge => (
-                  <span
-                    className={badgeInputContents}
-                    key={`${badge.value}-badgeWrapper`}
-                  >
-                    <Badge appearance={badgeAppearance}>
-                      <Flex align="center" gutterSize="xxs">
-                        <FlexItem>
-                          <div className={cx(textTruncate, badgeText)}>
-                            {typeof badge.label === "string"
-                              ? badge.label.replace(/\s/g, "\u00a0")
-                              : badge.label}
-                          </div>
-                        </FlexItem>
-                        <FlexItem flex="shrink">
-                          <Clickable
-                            action={this.onBadgeClose.bind(this, badge)}
-                          >
-                            <span>
-                              <Icon shape={SystemIcons.Close} size="xxs" />
-                            </span>
-                          </Clickable>
-                        </FlexItem>
-                      </Flex>
-                    </Badge>
-                  </span>
-                ))}
+                this.props.badges.map(badge => {
+                  const { badgeAppearance = "primary" } = this.props;
```

Comment:
we're already destructuring `this.props` above. Should we just do all of these there? unsure why this would make a difference

Reply:
Thought I was still seeing the error with it there while troubleshooting, but it looks likes what I've done elsewhere resolves that. Thanks!